### PR TITLE
Fix entry point selector

### LINKED ISSUE #180: Implement entry point selector

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fix some mistakes in the entry point selector.
Closes #180 

# Test Plan:
Added test

### CODE REVIEW

File: consensus/entry_point_selector/entry_point_selector.c [hunk 1]
```diff
@@ -26,27 +26,31 @@ retcode_t iota_consensus_entry_point_selector_init(
...
-retcode_t iota_consensus_get_entry_point(entry_point_selector_t *const eps,
-                                         size_t const depth,
-                                         trit_array_p const ep) {
+retcode_t iota_consensus_entry_point_selector_get_entry_point(
+    entry_point_selector_t *const eps, size_t const depth,
+    trit_array_p const ep) {
   retcode_t ret = RC_OK;
-  size_t milestone_index = MAX(eps->mt->latest_milestone_index - depth - 1, 0);
+  uint64_t milestone_index =
+      MAX(eps->mt->latest_solid_subtangle_milestone_index - depth - 1, 0);
```
Comment (th0br0) [inline consensus/entry_point_selector/entry_point_selector.c]: ~I believe you're missing a cast to `int64_t` here?~ You're not, but we probably shouldn't use `size_t` for `latest_solid_subtangle_milestone_index`?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: consensus/entry_point_selector/entry_point_selector.c (hunk 1 above)
Comment (thibault-martinez) [reply consensus/entry_point_selector/entry_point_selector.c]: `latest_solid_subtangle_milestone_index` is an `uint64_t`

Comment (th0br0) [reply consensus/entry_point_selector/entry_point_selector.c]: Ah it's not on the `get_node_info` struct - unrelated.
So yeah, you definitely want to cast here?

Comment (tsvisabo) [reply consensus/entry_point_selector/entry_point_selector.c]: could use `ULLONG_MAX`

**Review by tsvisabo — CHANGES_REQUESTED** [review]

**Review by tsvisabo — APPROVED** [review]

File: consensus/entry_point_selector/entry_point_selector.h [hunk 2]
```diff
@@ -27,14 +27,42 @@ typedef struct entry_point_selector_s {
...
+ * Given a depth N, returns a N-deep milestone, i.e. (latest_solid - depth)th
+ * milestone.
+ *
+ * @param eps The entry point selector
+ * @param depth The depth
+ * @param ep The entry point
+ *
+ * @return a status code
+ */
+extern retcode_t iota_consensus_entry_point_selector_get_entry_point(
```
Comment (th0br0) [inline consensus/entry_point_selector/entry_point_selector.h]: Why the explicit extern? Because there can be multiple impls?

Comment (thibault-martinez) [reply consensus/entry_point_selector/entry_point_selector.h]: No specific reason, I saw it many times elsewhere and somehow thought it was part of the coding style, I'll remove.

Comment (th0br0) [reply consensus/entry_point_selector/entry_point_selector.h]: The motivation for that is link/build-time pluggable implementations. Might make sense here in the future as well, we'll see.

**Review by th0br0 — APPROVED** [review]

**Pull request merged**